internal/services/storage/storage_container_resource.go

```diff
@@ -103,54 +109,59 @@ func storageContainerCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 	accessLevelRaw := d.Get("container_access_type").(string)
 	accessLevel := expandStorageContainerAccessLevel(accessLevelRaw)
 
+	metaDataRaw := d.Get("metadata").(map[string]interface{})
+	metaData := ExpandMetaData(metaDataRaw)
+
 	account, err := storageClient.FindAccount(ctx, accountName)
 	if err != nil {
-		return fmt.Errorf("retrieving Account %q for Container %q: %s", accountName, containerName, err)
+		return diag.Errorf("retrieving Account %q for Container %q: %s", accountName, containerName, err)
 	}
 	if account == nil {
-		return fmt.Errorf("Unable to locate Storage Account %q!", accountName)
+		return diag.Errorf("Unable to locate Storage Account %q!", accountName)
 	}
 
 	client, err := storageClient.ContainersClient(ctx, *account)
 	if err != nil {
-		return fmt.Errorf("building storage client: %+v", err)
+		return diag.Errorf("building storage client: %+v", err)
 	}
 
 	id := parse.NewStorageContainerDataPlaneId(accountName, storageClient.Env.StorageEndpointSuffix, containerName).ID()
 	exists, err := client.Exists(ctx, account.ResourceGroup, accountName, containerName)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 	if exists != nil && *exists {
-		return tf.ImportAsExistsError("azurestack_storage_container", id)
+		return diag.FromErr(tf.ImportAsExistsError("azurestack_storage_container", id))
 	}
 
 	log.Printf("[INFO] Creating Container %q in Storage Account %q", containerName, accountName)
```

Comment:
Using the functions of tf SDK 2+, this kind of logging is not necessary.

Reply:
this is not for logging, is just to return the diag.Diagnostic instead of error, as the tf SDK v2 upgrade guide says

Reply:
My bad, I meant for the line 137.